components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/AESDecryptor.java

```diff
@@ -0,0 +1,49 @@
+package org.wso2.carbon.identity.sso.agent;
+
+import org.apache.commons.codec.binary.Base64;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Class for decrypt the encrypted values.
+ *
+ */
+public class AESDecryptor {
+
+    private static final String ALGORITHM = "AES/CBC/PKCS5Padding";
+    private static final Logger LOGGER = Logger.getLogger(SSOAgentConstants.LOGGER_NAME);
+
+    public static String decode(String encryptedSecret, String cipherKey) {
+
+        SecretKey key = new SecretKeySpec(cipherKey.getBytes(StandardCharsets.UTF_8), "AES");
+        try {
+            Cipher cipher = Cipher.getInstance(ALGORITHM);
+            byte[] iv = new byte[cipher.getBlockSize()];
+
+            IvParameterSpec ivParams = new IvParameterSpec(iv);
+
+            cipher.init(Cipher.DECRYPT_MODE, key, ivParams);
+            byte[] decodedValue = new Base64().decode(encryptedSecret.getBytes(StandardCharsets.UTF_8));
+
+            byte[] decryptedValue = cipher.doFinal(decodedValue);
+
+            return new String(decryptedValue, StandardCharsets.UTF_8);
+        } catch (NoSuchPaddingException | NoSuchAlgorithmException | InvalidKeyException |
+                InvalidAlgorithmParameterException | IllegalBlockSizeException | BadPaddingException ex) {
+            LOGGER.log(Level.SEVERE, ex.getMessage() + ex);
```

Comment:
Is there any point in handling this exception at this point ? Cant we just throw them ?